addons/stock_account/models/stock_move.py

```diff
@@ -508,10 +508,15 @@ def _get_in_svl_vals(self, forced_quantity):
         return svl_vals_list
 
     def _get_src_account(self, accounts_data):
-        return self.location_id.valuation_out_account_id.id or accounts_data['stock_input'].id
+        if self.location_id.usage == 'inventory':
+            return self.location_id.valuation_in_account_id.id or accounts_data['income'].id
+        else:
+            return self.location_id.valuation_out_account_id.id or accounts_data['stock_input'].id
 
     def _get_dest_account(self, accounts_data):
-        if not self.location_dest_id.usage in ('production', 'inventory'):
+        if self.scrapped or self.location_dest_id.usage == 'inventory':
```

Comment:
In case you scrap a product into an internal location (in an open receipt for instance, the stock move will be 
Vendor -> WH/Stock/scrap). It will be valuated and should be considered as a regular receipt (and thus return 'stock_output')